src/mbgl/util/mapbox.cpp

```diff
@@ -187,6 +187,20 @@ std::string canonicalizeTileURL(const std::string& url, SourceType type, uint16_
     }
 
     result += "." + extension;
+
+    // get the query and remove access_token, if more parameters exist, add them to the final result
+    if (queryIdx != url.length()) {
+        const auto query = url.substr(queryIdx);
+        // get access token index
+        const auto accessTokenIdx = (query.find("access_token") == 0) ? 0 : query.find("access_token") - 1; // if it's not the first param, grab the & as well
+        const auto accessTokenEndIdx = (accessTokenIdx == 0) ? query.find("&", accessTokenIdx) + 1 : query.find("&", accessTokenIdx);
```

Comment:
This line assumes there is a second parameter by checking for `query.find("&", accessTokenIdx)`. If the response from this is `0`, we need to grab the end of the query for `accessTokenEndIdx`.